### PR TITLE
[24415] Update NetworkProvider implementation

### PR DESCRIPTION
**Pull request by mejiagarcia**
## Ticket
- [24415]

## Description
- `NetworkProvider` class improved a bit to improve readability.
- `Response` converted to `NetworkResponse` and made it public so an external `NetworkProvider` can be created.
- `FailureResponse`  also made it public.
- Mentioned models unified with `NetworkProvider` in the same file.
- `URLSession` extension moved to `NetworkProvider` file in order to centralize provider implementation.

## Commit message
-  fix: Make NetworkProvider's models public to allow creating a custom provider

## Testing
- [x] Added unit tests
- [x] Ran unit tests successfully
- [ ] Added documentation for public APIs and/or Wiki

### CODE REVIEW

File: TwilioVerifySDK/TwilioVerify/Sources/Networking/NetworkProvider.swift [hunk 1]
```diff
@@ -19,38 +19,186 @@
...
+    success: @escaping SuccessResponseBlock,
+    failure: @escaping FailureBlock
+  )
 }
 
-class NetworkAdapter: NetworkProvider {
-  
+// MARK: - Implementation
+
+final class NetworkAdapter: NetworkProvider {
```
Comment (yafuquen) [inline TwilioVerifySDK/TwilioVerify/Sources/Networking/NetworkProvider.swift:39]: Can we move this to a new file? Including the URLSession extension

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`93.56%`**
File | Coverage
-----|-----
TwilioVerifySDK/TwilioVerify/Sources/Networking/NetworkProvider.swift | **`93.40%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: TwilioVerifySDK/TwilioVerify/Sources/Networking/NetworkProvider.swift (hunk 1 above)
Comment (mejiagarcia) [reply TwilioVerifySDK/TwilioVerify/Sources/Networking/NetworkProvider.swift:22]: Done on last commit

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">TwilioVerifySDK/TwilioVerify/Sources/Networking/NetworkProvider+Models.swift has less than 70% code coverage</td>
    </tr>
  </tbody>
</table>

## Code coverage
### Total coverage: **`93.56%`**
File | Coverage
-----|-----
TwilioVerifySDK/TwilioVerify/Sources/Networking/NetworkProvider+Models.swift | **`64.29%`**
TwilioVerifySDK/TwilioVerify/Sources/Networking/NetworkProvider.swift | **`97.83%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">TwilioVerifySDK/TwilioVerify/Sources/Networking/NetworkProvider+Models.swift has less than 70% code coverage</td>
    </tr>
  </tbody>
</table>

## Code coverage
### Total coverage: **`93.55%`**
File | Coverage
-----|-----
TwilioVerifySDK/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`78.12%`**
TwilioVerifySDK/TwilioVerify/Sources/Networking/NetworkProvider+Models.swift | **`64.29%`**
TwilioVerifySDK/TwilioVerify/Sources/Networking/NetworkProvider.swift | **`97.83%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`93.74%`**
File | Coverage
-----|-----
TwilioVerifySDK/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`78.12%`**
TwilioVerifySDK/TwilioVerify/Sources/Networking/NetworkProvider+Models.swift | **`100.00%`**
TwilioVerifySDK/TwilioVerify/Sources/Networking/NetworkProvider.swift | **`97.83%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: TwilioVerifySDK/TwilioVerify/Sources/Networking/NetworkProvider.swift [hunk 2]
```diff
@@ -19,38 +19,117 @@
...
-public typealias SuccessResponseBlock = (Response) -> ()
-public typealias FailureBlock = (Error) -> ()
-
-public protocol NetworkProvider {
-  func execute(_ urlRequest: URLRequest, success: @escaping SuccessResponseBlock, failure: @escaping FailureBlock)
-}
-
-class NetworkAdapter: NetworkProvider {
-  
+final class NetworkAdapter: NetworkProvider {
```
[the remark this answers is not in the document]
Comment (mejiagarcia) [reply TwilioVerifySDK/TwilioVerify/Sources/Networking/NetworkProvider.swift]: Done on last commit.

**Review by AlejandroOrozco — APPROVED** [review]
LG 👍

**Review by yafuquen — APPROVED** [review]

**Pull request merged**